### DISCUSSION TITLE
MySql seems to be not supported.

### DISCUSSION THREAD

**Discussion by sjmittal** [ory/kratos#1653] [Q&A] [answered]
**Describe the bug**
When I run kratos version `v0.7.1-alpha.1` on kubernetes it does not seem to apply the migration

**Reproducing the bug**

In my kratos config I have defined my `dsn` as:
`mysql://user:secret@tcp(mysqld:3306)/database`

When the kratos pod is run I get following logs:
`
{"audience":"application","level":"debug","msg":"Ignoring migration file 20210618103120000013_logout_token.sqlite3.down.sql because dialect is not supported: unsupported dialect sqlite3","service_name":"Ory Kratos","service_version":"v0.7.1-alpha.1","time":"2021-08-16T06:32:31Z"}
`

Why is it choosing `sqlite3` dialect even when in dsn I have mentioned to connect to `mysql`.

Is there some other place I need to specify the dialect?

**Comment by sjmittal** [comment]
Can anyone help me with this. I checked the kratos code and I am unable to figure out from what part of the code this message is originating. Right now pod is not running because the schema is not getting created because looks like these migration scripts are not getting applied.

**Reply by aeneasr** [reply]
You can ignore this warning.

**Reply by sjmittal** [reply]
But bottom line is that no schema has got created and pod is not in running state.
The error I get is:
`{"audience":"application","error":{"message":"mysql select one: Error 1146: Table 'kratosdb_68cbce5a.networks' doesn't exist","trace":"\ngithub.com/ory/x/sqlcon.HandleError\n\t/go/pkg/mod/github.com/ory/x@v0.0.262/sqlcon/error.go:88\ngithub.com/ory/x/networkx.(*Manager).Determine\n\t/go/pkg/mod/github.com/ory/x@v0.0.262/networkx/manager.go:42\ngithub.com/ory/kratos/persistence/sql.(*Persister).DetermineNetwork\n\t/home/ory/persistence/sql/persister.go:77\ngithub.com/ory/kratos/driver.(*RegistryDefault).Init.func1\n\t/home/ory/driver/registry_default.go:535\ngithub.com/cenkalti/backoff.RetryNotify\n\t/go/pkg/mod/github.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:37\ngithub.com/cenkalti/backoff.Retry\n\t/go/pkg/mod/github.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:24\ngithub.com/ory/kratos/driver.(*RegistryDefault).Init\n\t/home/ory/driver/registry_default.go:475\ngithub.com/ory/kratos/driver.New\n\t/home/ory/driver/factory.go:14\ngithub.com/ory/kratos/cmd/serve.NewServeCmd.func1\n\t/home/ory/cmd/serve/root.go:33\ngithub.com/spf13/cobra.(*Command).execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:856\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960\ngithub.com/spf13/cobra.(*Command).Execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897\ngithub.com/ory/kratos/cmd.Execute\n\t/home/ory/cmd/root.go:48\nmain.main\n\t/home/ory/main.go:35\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371"},"level":"warning","msg":"Unable to determine network, retrying.","service_name":"Ory Kratos","service_version":"v0.7.1-alpha.1","time":"2021-08-18T11:51:35Z"}`

**Reply by aeneasr** [reply]
Are you running migrations?

**Reply by sjmittal** [reply]
Ok after running the migration all worked. Thanks for taking me through this.

**Reply by aeneasr** [reply]
My pleasure :)

**Comment by sjmittal** [comment, marked as answer]
Running the automigrate job in k8s fixed the issue.
